src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs

```diff
@@ -239,8 +239,34 @@ private static IEnumerable<string> GetPropertyArguments(IDictionary<string, ILis
 
         private static string EscapeSpecialCharacters(string value)
         {
-            return value.Replace(",", "%2C")
-                .Replace(";", "%3B");
+            if (string.IsNullOrEmpty(value))
+            {
+                return string.Empty;
+            }
+
+            var escapedBuilder = new StringBuilder();
```

Comment:
I think that a lookup dictionary would scale better here.

Reply:
Changed to dictionary, extracted to own class so it could be reused for the .NET Core msbuild stuff. Added a test.